pennylane/fourier/spectrum.py

```diff
@@ -0,0 +1,244 @@
+# Copyright 2018-2020 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Contains qnode transform that computes the fourier spectrum of a QNode."""
+import numpy as np
+import pennylane as qml
+from functools import wraps
+from itertools import product
+from copy import copy
+
+
+def _simplify_tape(tape, original_inputs):
+    r"""Expand the tape until every operation that takes an original input
+    is a single-parameter gate.
+
+    A single-parameter gate is of the form :math:`\exp(-i x_i G)` where :math:`x_i` is a scalar input
+    and :math:`G` is a Hermitian operator called a "generator".
+
+    If the expansion involves any processing on the parameters (such as scalar multiplication)
+    the circuit is not of a form that can be handled by the spectrum function, and an error is thrown.
+
+    Args:
+        tape (~.tape.QuantumTape): tape to simplify
+        original_inputs (list): list of inputs, to check that they are conserved throughout expansion
+
+    Returns:
+        ~.tape.QuantumTape: expanded tape
+    """
+
+    def stop_at(obj):
```

Comment:
Does this need to live inside the `_simplify_tape` function or can it be moved outside? It seems like it is most of the function.

Reply:
Oh yes, I forgot. Probably good to move it outside? I just thought the _simplify function is a good logical unit...

Let's worry about this in the final PR maybe!